src/runtime/virtcontainers/network_linux.go

```diff
@@ -211,13 +211,18 @@ func (n *LinuxNetwork) addSingleEndpoint(ctx context.Context, s *Sandbox, netInf
 	return endpoint, nil
 }
 
-func (n *LinuxNetwork) removeSingleEndpoint(ctx context.Context, s *Sandbox, idx int, hotplug bool) error {
+func (n *LinuxNetwork) removeSingleEndpoint(ctx context.Context, s *Sandbox, endpoint Endpoint, hotplug bool) error {
+	var idx int = len(n.eps)
+	for i, val := range n.eps {
+		if val == endpoint {
```

Comment:
Looking at [`findEndpoint`](https://github.com/kata-containers/kata-containers/blob/main/src/runtime/virtcontainers/endpoint.go#L232), we are relying on the endpoint hardware address to determine if two endpoints are the same. So here we should use
```
if val.HardwareAddr() == endpoint.HardwareAddr() {
```

Reply:
Thank you for your valuable feedback and suggestions. I have made the necessary modifications to address your concerns. You can review the changes in the updated.